cmd/skaffold/app/cmd/build.go

```diff
@@ -64,8 +67,19 @@ func doBuild(ctx context.Context, out io.Writer) error {
 	return withRunner(ctx, func(r runner.Runner, config *latest.SkaffoldConfig) error {
 		bRes, err := r.BuildAndTest(ctx, buildOut, targetArtifacts(opts, config))
 
-		if quietFlag {
+		if quietFlag || buildOutputFlag != "" {
 			cmdOut := flags.BuildOutput{Builds: bRes}
+
+			if buildOutputFlag != "" {
+				f, err := os.OpenFile(buildOutputFlag, os.O_RDWR|os.O_CREATE, 0755)
```

Comment:
Can you use ioutil.WriteFile like skaffold fix uses 
 here https://github.com/GoogleContainerTools/skaffold/blob/master/cmd/skaffold/app/cmd/fix.go#L73
`if err := ioutil.WriteFile(configFile, newCfg, 0644); err != nil {
}
`

Reply:
ah ok. 
Can you create a bytes.buffer and execute the template there
```
buildOut bytes.Buffer
t.Execute(buildOut, r)
```

And the if --quiet is present, 
```
out.Write(buildOut)
```
if "--file-output" is present, 
```
err := ioutil.WriteFile(buildOutputFlag, buildOut, 0644)
if err != nil {
  log.Fatal(err)
}
```
That ways its cleaner and we do not overwrite  `out`